managed/models/role_helpers.go

```diff
@@ -112,72 +76,6 @@ func DeleteRole(tx *reform.TX, roleID, replacementRoleID int) error {
 	return nil
 }
 
-func replaceRole(tx *reform.TX, roleID, newRoleID int) error {
-	q := tx.Querier
-
-	// If no new role is assigned, we remove all role assignements.
-	if newRoleID == 0 {
-		_, err := q.DeleteFrom(UserRolesView, "WHERE role_id = $1", roleID)
-		return err
-	}
-
-	if err := FindAndLockRole(tx, newRoleID, &Role{}); err != nil {
-		return err
-	}
-
-	// Check if the role is the last role for a user and apply special logic if it is.
-	structs, err := q.FindAllFrom(UserRolesView, "role_id", roleID)
-	if err != nil {
-		return err
-	}
-
-	for _, s := range structs {
-		ur, ok := s.(*UserRoles)
-		if !ok {
-			return fmt.Errorf("invalid data structure in user roles for role ID %d. Found %+v", roleID, s)
-		}
-
-		roleStructs, err := q.SelectAllFrom(UserRolesView, "WHERE user_id = $1 FOR UPDATE", ur.UserID)
-		if err != nil {
-			return err
-		}
-
-		// If there are more than 1 roles, we remove the role without a replacement.
-		if len(roleStructs) > 1 {
-			_, err := q.DeleteFrom(UserRolesView, "WHERE user_id = $1 AND role_id = $2", ur.UserID, roleID)
-			if err != nil {
-				return err
-			}
-
-			continue
-		}
-
-		// The removed role is the last one. We replace it with a new role.
-		_, err = q.Exec(`
-			UPDATE user_roles
-			SET
-				role_id = $1
-			WHERE
-				user_id = $2 AND
-				role_id = $3 AND
-				NOT EXISTS(
-					SELECT 1
-					FROM user_roles ur
-					WHERE
-						ur.user_id = $2 AND
-						ur.role_id = $1
-				)`,
-			newRoleID,
-			ur.UserID,
-			roleID)
-		if err != nil {
-			return err
-		}
-	}
-
-	return nil
-}
-
 // FindAndLockRole retrieves a role by ID and locks it for update.
 func FindAndLockRole(tx *reform.TX, roleID int, role *Role) error {
```

Comment:
in other places, we return the value, but here we pass it as one of the parameters. Any specific reason for inconsistency?